patches/node/fix_use_crypto_impls_for_compat.patch

```diff
@@ -0,0 +1,33 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shelley Vohr <shelley.vohr@gmail.com>
+Date: Wed, 12 Feb 2020 15:08:04 -0800
+Subject: fix: use crypto impls for compat
+
+BoringSSL does not export DSA_get0_q, ECDSA_SIG_get0_r, or ECDSA_SIG_get0_s. This
+patch works around that problem by using the implementations of those functions as
+found in the OpenSSL repo. I plan to upstream a version of this.
+
+diff --git a/src/node_crypto.cc b/src/node_crypto.cc
+index 41b72d010824234c567586524d22cae6ac849edb..b2403821bb3cab4aa4587fcb757eeb101c8a7e03 100644
+--- a/src/node_crypto.cc
++++ b/src/node_crypto.cc
+@@ -5027,7 +5027,7 @@ static unsigned int GetBytesOfRS(const ManagedEVPPKey& pkey) {
+   if (base_id == EVP_PKEY_DSA) {
+     DSA* dsa_key = EVP_PKEY_get0_DSA(pkey.get());
+     // Both r and s are computed mod q, so their width is limited by that of q.
+-    bits = BN_num_bits(DSA_get0_q(dsa_key));
++    bits = BN_num_bits(dsa_key->q);
```

Comment:
It looks like `dsa_key->q` would work for [boringssl](https://github.com/google/boringssl/blob/8728af480156e3471775a41ec33c352e0b09000d/include/openssl/dsa.h#L399)  but maybe not always for [openssl](https://github.com/openssl/openssl/blob/663247bf1181043d168a542b0d0d56cc0318e2f7/crypto/dsa/dsa_lib.c#L267)?

I wasn't sure from your comment about "plan to upstream a version of this" whether you're referring to upstreaming to Node or to another project -- IMO it might be worthwhile to see if boringssl would be agreeable to a patch that implements these helpers there

Reply:
this has been upstreamed and merged 😸